hotsos/core/plugins/kernel/net.py

```diff
@@ -220,3 +220,81 @@ def _fields(self):
             'TCPReqQFullDoCookies',
             # skb retrans before original left host
             'TCPSpuriousRtxHostQueues']
+
+
+class ProcSockStat():
+
+    """
+    Provides a common way to extract fields from /proc/net/sockstat.
+
+    Expected file format is:
+
+    hdr1: label1 value1 label2 value2...
+    hdr2: label1 value1 label2 value2...
+    hdr3: label1 value1 label2 value2...
+    """
+    def __init__(self):
+        self._data = {}
+        self._process_file(os.path.join(HotSOSConfig.DATA_ROOT,
+                                        'proc/net/sockstat'))
+
+    def _process_file(self, fname):
```

Comment:
I would be great if we could somehow adapt the baseclass to be able to cover both formats but i guess that can be done seperately.